src/panels/config/zha/zha-node.js

```diff
@@ -0,0 +1,542 @@
+  }
+
+  computeIsClusterAttributeSelected(selectedClusterAttribute) {
+    return this.clusters && selectedClusterAttribute !== -1;
+  }
+
+  computeIsClusterCommandSelected(selectedClusterCommand) {
+    return this.clusters && selectedClusterCommand !== -1;
+  }
+
+  computeNodeServiceData(selectedNode) {
+    return { ieee_address: this.nodes[selectedNode].attributes.ieee };
+  }
+
+  computeClusterServiceData(selectedNode, selectedEntity, selectedCluster) {
+    return {
+      ieee_address: this.nodes[selectedNode].attributes.ieee,
+      entity_id: this.entities[selectedEntity].entity_id,
+      cluster_id: this.clusters[selectedCluster].id,
+      cluster_type: this.clusters[selectedCluster].type,
+    };
+  }
+
+  computeClusterCommandServiceData(selectedClusterCommand) {
+    return {
+      entity_id: this.entities[this.selectedEntity].entity_id,
+      cluster_id: this.clusters[this.selectedCluster].id,
+      cluster_type: this.clusters[this.selectedCluster].type,
+      command: this.clusterCommands[selectedClusterCommand].id,
+      command_type: this.clusterCommands[selectedClusterCommand].type,
+    };
+  }
+
+  computeReadAttributeServiceData() {
+    return {
+      entity_id: this.entities[this.selectedEntity].entity_id,
+      cluster_id: this.clusters[this.selectedCluster].id,
+      cluster_type: this.clusters[this.selectedCluster].type,
+      attribute: this.clusterAttributes[this.selectedClusterAttribute].id,
+      manufacturer:
+        this.manufacturerCodeOverride != null
+          ? parseInt(this.manufacturerCodeOverride)
+          : this.nodes[this.selectedNode].attributes.manufacturer_code,
+    };
+  }
+
+  computeSetAttributeServiceData(attributeValue) {
+    return {
+      entity_id: this.entities[this.selectedEntity].entity_id,
+      cluster_id: this.clusters[this.selectedCluster].id,
+      cluster_type: this.clusters[this.selectedCluster].type,
+      attribute: this.clusterAttributes[this.selectedClusterAttribute].id,
+      value: attributeValue,
+      manufacturer:
+        this.manufacturerCodeOverride != null
+          ? parseInt(this.manufacturerCodeOverride)
+          : this.nodes[this.selectedNode].attributes.manufacturer_code,
+    };
+  }
+
+  selectedEntityChanged(selectedEntity) {
+    if (selectedEntity === -1) return;
+    this.selectedCluster = -1;
+    this.selectedClusterAttribute = -1;
+    this.selectedClusterCommand = -1;
+    this.hass
+      .callWS({
+        type: "zha/entities/clusters",
+        entity_id: this.entities[selectedEntity].entity_id,
+        ieee: this.entities[selectedEntity].device_info.identifiers[0][1],
+      })
+      .then((clusters) => {
+        this.setProperties({ clusters: clusters });
+      });
+  }
+
+  selectedClusterChanged(selectedCluster) {
+    if (selectedCluster === -1) return;
+    this.selectedClusterAttribute = -1;
+    this.selectedClusterCommand = -1;
+    this.hass
+      .callWS({
+        type: "zha/entities/clusters/attributes",
```

Comment:
Please move all these commands to `src/data/zha.ts`. See `src/data/shopping-list.ts` as an example.